### DISCUSSION TITLE
custom collection and endpoint for the collection

### DISCUSSION THREAD

**Discussion by sthisfake** [pocketbase/pocketbase#2677] [Q&A] [answered]
Hey, first of all, this is an awesome project you guys are working on. I have a question. I'm currently working on a project with some Python models, and I want to create an API based on the inputs and outputs of these models. I want to use PocketBase to create the API, but I'm struggling to create a custom collection and attach API methods for that collection that can run the Python script in the background.

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure that I understand your use case, but at the moment the only way to register a custom route/endpoint is to [use PocketBase as Go framework](https://pocketbase.io/docs/use-as-framework/), which will allow you to [register a new route](https://pocketbase.io/docs/router/#register-new-route) via the `OnBeforeServe` hook.

_With the next major release there will be also an option to bind to the app hooks with JavaScript via [goja](https://github.com/dop251/goja) (_there is a POC in the `develop` branch_)._

---

Alternatively, if you don't want to extend PocketBase with Go, you can use the prebuilt executable and have a Python middleware service that will interact with PocketBase using its [Web APIs](https://pocketbase.io/docs/api-records/) (you can use the web apis both as regular app user or admin; admins have access to everything and collection API rules don't apply for them).
